src/lib/hooks/useABTesting.js

```diff
@@ -1,21 +1,48 @@
-import { useEffect } from 'react'
+import { useEffect, useState } from 'react'
```

Comment:
Optimized persist feature

1. the whole list of persisted test is loaded once
2. reduced storage amount (persisting  isCaseA boolean flags only)
3. instead of exporting test promise `initialized`var was added to the hook's api. it could be used in useTorus - just add effect to its change instead of awaiting testPromise. This will be React way to wait for this

```javascript
import { useEffect, useState } from 'react'
import { once, isNil, pick } from 'lodash'

import Config from '../../config/config'
import { fireEvent } from '../analytics/analytics'

import AsyncStorage from '../utils/asyncStorage'
import { AB_TESTING } from '../constants/localStorage'

import logger from '../logger/pino-logger'

let persistedVariants
const log = logger.child({ from: 'useABTesting' })

const loadPersistedVariants = once(async () => {
  persistedVariants = (await AsyncStorage.getItem(AB_TESTING)) || {}  
})

const createABTesting = (testName, percentage = Config.abTestPercentage, persistVariant = true) => {
  const generateABVariant = () => Math.random() < percentage
  const getABValue = variant => variant ? 'A' : 'B'

  const getABVariant = once(async () => {
    let isCaseA = persistVariant ? null : generateABVariant()

    if (persistVariant) {      
      await loadPersistedVariants()

      if (!(testName in persistedVariants)) {
        persistedVariants[testName] = generateABVariant()
        AsyncStorage.setItem(AB_TESTING, persistedVariants)
      }
      
      isCaseA = persistedVariants[testName]
    }

    const ab = isCaseA ? 'A' : 'B'
    
    log.debug('got test variant', { testName, persistVariant, isCaseA, ab, persistedVariants })
    return { isCaseA, ab }
  })

  const useABTesting = (componentA, componentB, event = null) => {
    const [abVariant, setABVariant] = useState()
    const initialized = !isNil(abVariant)
    const component = initialized ? (abVariant.isCaseA ? componentA : componentB) : null

    useEffect(() => {
      getABVariant().then(variant => {
        if (event) {
          fireEvent(event, pick(variant, 'ab'))
        }

        setABVariant(variant)
        log.debug('hook ready', variant)
      })
    }, [])

    return [component, abVariant.ab, initialized]
  }

  return { useABTesting }
}

export default createABTesting

```

Reply:
refactored a little as your suggestions